### DISCUSSION TITLE
What is the difference between MakeAppx.exe and MakeMsix.exe?

### DISCUSSION THREAD

**Discussion by drauch** [Microsoft/msix-packaging#598] [Q&A] [answered]
Both seem to be able to generate MSIX files. Is it not recommended to pack MSIX files using MakeAppx.exe?

Best regards,
D.R.

**Comment by florelis** [comment, marked as answer]
MakeAppx.exe is part of the Windows SDK, closed source and only available on Windows.

MakeMsix.exe is a port of MakeAppx.exe intended to be cross-platform, so that it can also be compiled for, and run on, Linux and MacOS. The source code for it is available in this repository, however it does not have feature parity with MakeAppx.exe.

In general, it probably is better to use MakeAppx.exe if you can.

**Reply by DrusTheAxe** [reply]
+1

**Comment by drauch** [comment]
@florelis: This sounds like there will be a point in time when MakeMsix will be the new official tool and MakeAppx will be be legacy, am I right? :-) Or is MakeAppx still developed first and MakeMsix is always a step behind?

Best regards,
D.R.

**Reply by DrusTheAxe** [reply]
There's no current plans for MakeMsix to replace MakeAppx, nor to 'port' MakeAppx to non-Windows platforms.

They serve difference purposes today. Are there feature disparities you find problematic? If so please share your feedback.
